### PR TITLE
Fix bounce back bug in publisher configuration page

### LINKED ISSUE #677: [Bug][Frontend] Sometimes when you click on the Data Viz page and click on another link, it briefly takes you to your destination and suddenly snaps you back to you back to the Data Viz page
There is a strange intermittent bug when you click to go to the Data (visualization) page and go to another page - there are moments when you get redirected back to the Data page. It doesn't happen all the time and there's no clear way to trigger this. Some times it takes a few seconds before you get snapped back, sometimes it happens immediately, sometimes not at all. It feels like there's something funny going on with either the routing or some state that the Data page updating after you leave that page. I was able to capture a video of this behavior happening a couple of times:

https://github.com/Recidiviz/justice-counts/assets/59492998/e3f46adf-2536-4182-b7c4-3ff8e849012e

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> Fix bug where you get thrown back to data/config page after you switch to another page

## Type of change
Type: Bug

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #677 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by ivan-kishko** [comment]
Hello @lilidworkin @mxosman. So this is WIP PR because I only fixed metric config page. Let me explain the idea of how I decided to fix that and if you like it I will try to fix data page (explore data now) the same way. So basically what was causing the bug (I wrote about it in task itself) is that after loading data we change state also with setting search params if there are none because we needed like 1 source of truth for app to realize which system and metric to display. The only idea I came up for now is to remove this initial search param setting, and instead if there is no system param just display first from agency. Problem with that is that before logic that I removed was checking whether given system in search params belongs to agency (you know like if you share link) and if not we switch to first in agency as well (example would be like http://localhost:3000/agency/150/data?system=some_jibberish). So I decided if that is the case we just show Not Found Page and if user click "Set Up Metrics" link he goes to first system in agency and see those system metrics. Something similar can be done if system is in agency but metric is not (example http://localhost:3000/agency/161/metric-config?system=jails&metric=jails_jibberish)

**Comment by ivan-kishko** [comment]
Added show not found page if metric is not in system, also home button in not found redirected you to records page and since with new ui we have literally home button I fixed link to go there

File: publisher/src/components/MetricsConfiguration/MetricsConfiguration.tsx [hunk 1]
```diff
@@ -15,84 +15,32 @@
...
   const [loadingErrorMessage, setLoadingErrorMessage] = useState<string>();
 
   const initializeMetricConfiguration = async () => {
     setIsLoading(true);
-
     const response = await initializeMetricConfigStoreValues(agencyId);
     if (response instanceof Error) {
       return setLoadingErrorMessage(response.message);
     }
-
```
Comment (alfonsogoberjr) [inline publisher/src/components/MetricsConfiguration/MetricsConfiguration.tsx:50]: So this logic was completely removed?

**Review by alfonsogoberjr — APPROVED** [review]

**Comment by alfonsogoberjr** [comment]
@mxosman or @lilidworkin should also approve before merge

**Comment by mxosman** [comment]
> Hello @lilidworkin @mxosman. So this is WIP PR because I only fixed metric config page. Let me explain the idea of how I decided to fix that and if you like it I will try to fix data page (explore data now) the same way. 

> So basically what was causing the bug (I wrote about it in task itself) is that after loading data we change state also with setting search params if there are none or system/metric is not in agency/system because we needed like 1 source of truth for app to realize which system and metric to display. 

> The only idea I came up for now is to remove this initial search param setting logic, and instead if there is no system param just display first from agency. Problem with that is that before logic that I removed was checking whether given system in search params belongs to agency (you know like if you share link) 
and if not we switch to first system in agency as well (example would be like http://localhost:3000/agency/150/data?system=some_jibberish switch to http://localhost:3000/agency/150/data?system=legit_system). 

> So I decided if that is the case we just show Not Found Page and if user click "Set Up Metrics" link he goes to first system in agency and see those system metrics. Something similar can be done if system is in agency but metric is not (example http://localhost:3000/agency/161/metric-config?system=jails&metric=jails_jibberish)

Hi Ivan - thank you for this recap and the explanation! I actually really like this solution. I'm personally OK with us showing a Not Found page if the query params are wrong vs. trying to correct the URL with the side effect of having this bounce back. 

cc'ing @lilidworkin to also get her thoughts - are you OK with the UI showing the NotFound page if the user has a typo in the query params (or the query params have a system they don't belong to) instead of redirecting to the first system? We shouldn't have issues internally with users getting to where they need to go and the query params being right (unless we have a bug) - but, I could see issues come up as Ivan pointed out w/ typos when sharing a link (or someone clicking on a link that has the `system` set to one they don't belong to).

> Added show not found page if metric is not in system, also home button in not found redirected you to records page and since with new ui we have literally home button I fixed link to go there

Nice touch! Thank you!

File: publisher/src/components/MetricsConfiguration/MetricsConfiguration.tsx (hunk 1 above)
Comment (mxosman) [reply publisher/src/components/MetricsConfiguration/MetricsConfiguration.tsx:50]: Yeah - this is the logic that had that intermittent bounce back effect if you went to the Metric Settings page and another page very quickly (example https://github.com/Recidiviz/justice-counts/issues/677).

**Review by mxosman — APPROVED** [review]
Thank you, Ivan! Will give @lilidworkin a chance to review as well.

**Comment by ivan-kishko** [comment]
I guess I merge this part and make a separate task for fixing this on data page since here it is more difficult, and I am still not sure why

**Review by lilidworkin — APPROVED** [review]
@ivan-kishko thanks for the thoughtful explanation! Your solution sounds totally fine with me. And agreed, we can merge this in and you can work on fixing on the data page separately. Thanks!

**Pull request merged**